File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Loading/AtomCircularProgressIndicator.swift
import SwiftUI

public struct AtomCircularProgressIndicator: View {
    @Environment(\.theme) private var theme

    private let iconSize: SizingRoles.Icon
    private let colorRole: IconColorRole

    public init(iconSize: SizingRoles.Icon = .m, colorRole: IconColorRole = .primary) {
        self.iconSize = iconSize
        self.colorRole = colorRole
    }

    public var body: some View {
        ProgressView()
            .tint(colorRole.color(from: theme))
            .frame(width: iconSize.rawValue, height: iconSize.rawValue)
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/loading/AtomCircularProgressIndicator.kt
package com.perrystreet.woof.designsystem.atomic.atoms.loading

import androidx.compose.foundation.layout.size
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.sizing.SizingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.icon.roles.IconColorRole
import com.perrystreet.woof.designsystem.theme.Theme

@Composable
fun AtomCircularProgressIndicator(
    iconSize: SizingRoles.Icon = SizingRoles.Icon.M,
    colorRole: IconColorRole = IconColorRole.Primary,
) {
    CircularProgressIndicator(
        modifier = Modifier.size(iconSize.dp),
        color = colorRole.color(),
        strokeWidth = SizingRoles.HorizontalRule.S.dp,
        trackColor = Theme.colors.outlineVariant,
    )
}
